specs/PushNotifications/PushNotifications-spec.md

```diff
@@ -0,0 +1,438 @@
+            byte[] Payload { get; };
+
+            // Takes a deferral to run under specific modes like low power mode
+            void GetDeferral();
+
+            // Closes the Deferral taken on a Push Trigger. If this is not called by the developer, the deferral will automatically be invoked on destruction.
+            void CompleteDeferral();
+
+            // Subscribe to Cancelled event handler to be signalled when resource policies are no longer true like 30s wallclock timer
+            event Windows.ApplicationModel.Background.BackgroundTaskCanceledEventHandler Canceled;
+
+            // The number of times Resource management suspended the Task
+            UInt32 SuspendedCount { get; };
+        }
+
+        // An abstraction over the inproc activation Registration flow
+        runtimeclass InProcActivatorDetails
+        {
+            // Initialize using a manifest defined activatorId to Activate the InProc COM server
+            InProcActivatorDetails(String taskClsid);
+
+            // Initialize an inproc activator for LOW-IL apps like UWP with no COM Server specified in manifest
+            InProcActivatorDetails();
+
+            // The CLSID associated with the Client COM server that reunion will adopt
+            String TaskClsid{ get; };
+
+            // The conditions under which Push Triggers would execute
+            Windows.ApplicationModel.Background.IBackgroundCondition Condition;
+
+            // The in-memory Event handler to subscribe to if the app is in the foreground
+            event Windows.Foundation.EventHandler<PushNotificationActivatedEventArgs> PushActivated;
+
+            // The backgroundRegistration that gets populated on successful registration
+            Windows.ApplicationModel.Background.IBackgroundTaskRegistration Registration;
+        };
+
+        enum ChannelStatus
+        {
+            InProgress, // The request is in progress and there is no retry operation
+            InProgressRetry, // The request is in progress and is in a backoff retry state
+            CompletedSuccess, // The request completed successfully
+            CompletedFailure, // The request failed with some critical internal error. Check ExtendedError for HRESULT
+        };
+
+        runtimeclass PushNotificationChannelResult
+        {
+            PushNotificationChannelResult(
+                Windows.Networking.PushNotifications.PushNotificationChannel channel,
+                HRESULT extendedError,
+                ChannelStatus status);
+
+            // The Push channel associated with the Result. Null if InProgress or completion failed
+            Windows.Networking.PushNotifications.PushNotificationChannel Channel { get; };
+
+            // Low-level information about why the operation might not
+            // have worked. Apps generally do not take action based on
+            // this value, in part because the value may not be stable.
+            HRESULT ExtendedError{ get; };
```

Comment:
What does "may not be stable" mean? If it's not guaranteed to be useful or may be wrong at any time, I'd think about deleting this?

Reply:
Yeah this needs to be reworded. Its essentially an error code that we surface to the developer. Mostly for logging. The State is more important.

Reply:
Suggestion: "More detailed error code." and leave it at that

Regardless the wording you pick I'm sure you'll get feedback when you go through API Review :-)

Reply:
Thanks @DrusTheAxe . I will go ahead and leave it as you suggested.